src/service/viewer-impl.js

```diff
@@ -251,7 +268,7 @@ export class Viewer {
     this.hasBeenVisible_ = this.isVisible();
```

Comment:
This line has to be called after `this.recheckVisibilityState_()` call, otherwise it will misreport. Probably just better to call `recheckVisibilityState_()` as soon as `viewerVisibilityState_` and `docState_` are initialized?